src/control/system/faultdomain.go

```diff
@@ -239,6 +228,30 @@ func (t *FaultDomainTree) WithNodeDomain(domain *FaultDomain) *FaultDomainTree {
 	return t
 }
 
+// WithID changes the integer ID of the FaultDomainTree node.
+func (t *FaultDomainTree) WithID(id uint32) *FaultDomainTree {
+	if t == nil {
+		t = NewFaultDomainTree()
+	}
+	t.ID = id
+	return t
+}
+
+// nextID walks the tree to figure out what the next unique ID is
```

Comment:
That sounds wise. I don't think we can guarantee we're never accessing it concurrently. In fact probably all the modification operations on a particular tree should be locked -- could have some weird and surprising results if one thread tries to merge a new domain into a branch while another removes the branch at a higher level.

Reply:
Removed the possibility for direct access to the member DB copy of the tree outside the DB. With that I don't think it is possible for anyone to read or write to the tree without going through the Database structure, which synchronizes access.

I did add a note about thread safety in a comment on the structure itself, in case we have other uses for it in the future.